students/AndyKwok/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,81 @@
+# Description: Mailroom Program (Database Organizer)
+# Author: Andy Kwok
+# Last Updated: 07/23/2018
+# ChangeLog: 
+# v1.0 - Initialization
+# v2.0 - Adopt file creation and dictionary types
+
+#!/usr/bin/env python3
+
+if __name__ == '__main__':    
+    database = {'Donor A': [2300.30, 1000.01, 2000.00],
+                'Donor B': [1394.12],
+                'Donor C': [14349.00, 175.95],
+                'Donor D': [42800.11, 1500.04, 3534.20],
+                'Donor E': [12312.23, 250.50]}
+    
+def thank_you_select(database):
+    """ Function
+    """
+    entry = input('Enter donor\'s name to search or \'list\' to show all donors> ')
+    if entry == 'list':
+        database_print(database)
+    else:
+        database = add_donation(entry, database)
+
+def report_select(database):
+    """ Function to sort the database and printout the sorted database
+    """
+    database = dict(sorted(database.items(), key=lambda x: x[1], reverse=True))
+    database_print(database)
+
+def letters_everyone(database):
+    for k, v in database.items():
+        file = open(k.replace(" ", "_") + ".txt", "w")
+        file.write(send_thank_you(k, v))
```

Comment:
it is preferred to use `with` statement when working with files and `file` is a reserved keyword in python (notice special highlighting here)